src/main/java/com/aws/greengrass/status/FleetStatusService.java

```diff
@@ -528,118 +653,145 @@ private void uploadFleetStatusServiceData(Set<GreengrassService> greengrassServi
         }
 
         if (!isConnected.get() && !Trigger.isCloudDeploymentTrigger(trigger)) {
-            logger.atDebug().log("Not updating fleet status data since MQTT connection is interrupted");
+            logger.atDebug().log(
+                    "Not updating fleet status data since MQTT connection is interrupted");
             return;
         }
         List<ComponentDetails> components = new ArrayList<>();
 
-        //When a component version is bumped up, FSS may have pointers to both old and new service instances
-        //Filtering out the old version and only sending the update for the new version
+        // When a component version is bumped up, FSS may have pointers to both old and new service
+        // instances
+        // Filtering out the old version and only sending the update for the new version
         Set<GreengrassService> filteredServices = new HashSet<>();
-        greengrassServiceSet.forEach(service -> {
-            try {
-                GreengrassService runningService = kernel.locate(service.getName());
-                filteredServices.add(runningService);
-            } catch (ServiceLoadException e) {
-                //not able to find service, service might be removed.
-                filteredServices.add(service);
-            }
-        });
+        greengrassServiceSet.forEach(
+                service -> {
+                    try {
+                        GreengrassService runningService = kernel.locate(service.getName());
+                        filteredServices.add(runningService);
+                    } catch (ServiceLoadException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Exception handling issues detected where catch blocks only return values, log messages, or record metrics without properly handling the error. This can cause silent failures where errors go unnoticed, leading to incorrect results and making it hard to debug problems in production. To fix this, make sure catch blocks include proper logging with the full exception object, re-throw exceptions to let callers handle them, or add fallback mechanisms like backup services. Learn more about proper exception handling at https://owasp.org/www-community/Improper_Error_Handling